src/User-repo.js

```diff
@@ -1,46 +1,62 @@
 class UserRepo {
   constructor(users) {
     this.users = users;
-  };
+  }
+  
   getDataFromID(id) {
     return this.users.find((user) => id === user.id);
-  };
+  }
+
   getDataFromUserID(id, dataSet) {
     return dataSet.filter((userData) => id === userData.userID);
-  };
+  }
+
   calculateAverageStepGoal() {
-    var totalStepGoal = this.users.reduce((sumSoFar, data) => {
-      return sumSoFar = sumSoFar + data.dailyStepGoal;
+    var totalStepGoal = this.users.reduce((total, data) => {
+      return total = total + data.dailyStepGoal;
     }, 0);
     return totalStepGoal / this.users.length;
-  };
+  }
+
   makeSortedUserArray(id, dataSet) {
     let selectedID = this.getDataFromUserID(id, dataSet)
     let sortedByDate = selectedID.sort((a, b) => new Date(b.date) - new Date(a.date));
     return sortedByDate;
   }
+
   getToday(id, dataSet) {
     return this.makeSortedUserArray(id, dataSet)[0].date;
-  };
+  }
+
   getFirstWeek(id, dataSet) {
     return this.makeSortedUserArray(id, dataSet).slice(0, 7);
-  };
+  }
+
+  getDateIndex(date, id, dataSet) {
+    return this.makeSortedUserArray(id, dataSet)
+      .indexOf(this.makeSortedUserArray(id, dataSet)
```

Comment:
First time with chaining. This is a neat feature for JS, glad it passes the test.